### PR TITLE
added code for handling hermitian  = True for pinv linear operation in pytorch frontend

### LINKED ISSUE #22793: <Add Linear Algebra operations to PyTorch frontend >
-[pinv] #7479

### PR DESCRIPTION
**Pull request by Monsurat-Onabajo**
# PR Description 

this pull request was created to solve this issue : #22793
## Related Issue 

Close #22793
https://github.com/unifyai/ivy/issues/22793

## Checklist 

- [✅] Did you add a function?
- [✅] Did you add the tests?
- [✅] Did you follow the steps we provided?

### Socials: 

[@onabajo3](https://twitter.com/Onabajo3)

### CODE REVIEW

**Comment by Monsurat-Onabajo** [comment]
@Mr-Niraj-Kulkarni Can you review my pull request please

File: ivy_tests/test_ivy/test_frontends/test_torch/test_linalg.py [hunk 1]
```diff
@@ -88,6 +88,312 @@ def _get_dtype_and_matrix(
...
+        frontend=frontend,
+        test_flags=test_flags,
+        fn_tree=fn_tree,
+        on_device=on_device,
+        input=x[0],
+        atol=1e-02,
+        rtol=1e-02,
+        hermitian=hermitian
+    )
+
```
Comment (Mr-Niraj-Kulkarni) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_linalg.py:224]: Please add only parts specific to your implemented function

**Comment by Mr-Niraj-Kulkarni** [comment]
@Monsurat-Onabajo Hey, I added some comments on the code Please address those Also I see the tests are failing can you please look into that as well?  There are lint issues as well please resolve them

**Comment by Monsurat-Onabajo** [comment]
@Mr-Niraj-Kulkarni I have worked on the test reviews. I have also run pre-commit tests to solve the linting errors but i noticed that some of the errors that are coming from there are not from my contribution.

File: ivy_tests/test_ivy/test_frontends/test_torch/test_linalg.py [hunk 2]
```diff
@@ -875,13 +875,7 @@ def test_torch_multi_dot(
...
 )
 def test_torch_pinv(
-    *,
-    dtype_and_input,
-    frontend,
-    test_flags,
-    fn_tree,
-    backend_fw,
-    on_device,
+    *, dtype_and_input, frontend, test_flags, fn_tree, backend_fw, on_device, hermitian
```
Comment (zhumakhan) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_linalg.py:878]: Please draw the hermitian value from random boolean value generator like below:
```
@handle_frontend_test(
    fn_tree="torch.linalg.pinv",
    dtype_and_input=_get_dtype_and_matrix(batch=True),
    hermitian=st.booleans(),
)
def test_torch_pinv(
    *,
    dtype_and_input,
    frontend,
    test_flags,
    fn_tree,
    backend_fw,
    on_device,
    hermitian,
):
    input_dtype, x = dtype_and_input
    helpers.test_frontend_function(
        input_dtypes=input_dtype,
        backend_to_test=backend_fw,
        frontend=frontend,
        test_flags=test_flags,
        fn_tree=fn_tree,
        on_device=on_device,
        input=x[0],
        atol=1e-02,
        rtol=1e-02,
        hermitian=hermitian
    )
```

and please fix the test failures that appear after that

**Review by zhumakhan — CHANGES_REQUESTED** [review]

**Comment by Monsurat-Onabajo** [comment]
Hi, @Mr-Niraj-Kulkarni ...I was away on a conference and  I did not have access to my computer but i am back now. I tried solving the tasks but there is a circular import that is conflicting with the file i am trying to run, I kept on running into this error. "
ImportError: cannot import name 'array_helpers_dtype_info_helper' from partially initialized module 'ivy_tests.test_ivy.helpers.hypothesis_helpers.array_helpers' (most likely due to a circular import) (C:\Users\Onabajo Monsurat\Documents\venv\lib\site-packages\ivy_tests\test_ivy\helpers\hypothesis_helpers\array_helpers.py)"

**Comment by Mr-Niraj-Kulkarni** [comment]
> Hi, @Mr-Niraj-Kulkarni ...I was away on a conference and I did not have access to my computer but i am back now. I tried solving the tasks but there is a circular import that is conflicting with the file i am trying to run, I kept on running into this error. " ImportError: cannot import name 'array_helpers_dtype_info_helper' from partially initialized module 'ivy_tests.test_ivy.helpers.hypothesis_helpers.array_helpers' (most likely due to a circular import) (C:\Users\Onabajo Monsurat\Documents\venv\lib\site-packages\ivy_tests\test_ivy\helpers\hypothesis_helpers\array_helpers.py)"

Can you please try updating your branch from the main this might solve the issue

**Comment by Monsurat-Onabajo** [comment]
> > Hi, @Mr-Niraj-Kulkarni ...I was away on a conference and I did not have access to my computer but i am back now. I tried solving the tasks but there is a circular import that is conflicting with the file i am trying to run, I kept on running into this error. " ImportError: cannot import name 'array_helpers_dtype_info_helper' from partially initialized module 'ivy_tests.test_ivy.helpers.hypothesis_helpers.array_helpers' (most likely due to a circular import) (C:\Users\Onabajo Monsurat\Documents\venv\lib\site-packages\ivy_tests\test_ivy\helpers\hypothesis_helpers\array_helpers.py)"
> 
> Can you please try updating your branch from the main this might solve the issue

I have tried updating the main with my local clone but i am still getting the same error. When I checked, the error is coming from **'ivy_tests\test_ivy\helpers\hypothesis_helpers\array_helpers.py'** line 1513 and 1525 **'array_helpers_dtype_info_helper**' is referencing the backend implementation of it and line 1632 where **array_helpers_dtype_info_helper** is a function on its own, I am stuck because i dont want to change the name of the function so as not to break any code

**Review by Ishticode — CHANGES_REQUESTED** [review]
Hi @Monsurat-Onabajo 
Thanks for the PR and sorry for the delay on our end. I will take on your PR from now. Could you pls update your test code in accordance with what @zhumakhan had requested above i.e. add in a boolean strategy for hermitian and resolve the conflicts. Free free to request a further review and update after that. Thank you very much :)

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.

**Pull request closed without merging**